addons/packages/secretgen-controller/0.7.1/package.yaml

```diff
@@ -3,23 +3,41 @@ kind: Package
 metadata:
   name: secretgen-controller.community.tanzu.vmware.com.0.7.1
 spec:
+  valuesSchema:
+    openAPIv3:
+      type: object
+      additionalProperties: false
+      properties:
+        secretgenController:
+          type: object
```

Comment:
For properties of type object, we'll need to add `default: {}` so that openAPI utilities can use this to generate an object out of this.

Reply:
@vijaykatam , Did you mean something like below, check out default:
```
openapi: 3.0.0
info:
  version: 0.1.0
  title: Schema for data values, generated by ytt
paths: {}
components:
  schemas:
    dataValues:
      type: object
      additionalProperties: false
      description: OpenAPIV3 Schema for secret gen controller
      properties:
        secretgenController:
          type: object
          additionalProperties: false
          description: Configuration for secret gen controller
          properties:
            namespace:
              type: string
              default: secretgen-controller
              description: Namespace for secret gen controller
            createNamespace:
              type: boolean
              default: false
              description: Whether to create namespace for secret gen controller if not present
          default: {}
```

Reply:
I ask because default and properties is mutually exclusive according to carvel team and above is not valid schema. 
If something else is expected, let me know or if there is an example which has default for object type

Reply:
I haven't looked at openapi schema spec(https://spec.openapis.org/oas/v3.1.0) to validate if what you are saying is true but the reason I brought it up is because k8s apiextensions-server defaulting relies on non-nil default for an object, see https://github.com/kubernetes/apiextensions-apiserver/tree/master/pkg/apiserver/schema/defaulting

Reply:
Hi @shivaani0505 , please refer to the changes in https://github.com/vmware-tanzu/tanzu-framework/pull/1422/files#diff-8b8b2f2c353cbd029665d4e29754f2ed8d12ab749d2968431ed6563f113a01e4R47. It seems that Vijay handled this case in the util func and we don't need to add empty default objects.

Reply:
Sure @blc1996 . The test is blocking this MR. I have 2 options for testing listed above. We can discuss on those or a completely new one for testing

Reply:
Added all the testing tasks as discussed, so now closing this